src/libraries/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.TestService/TestServiceInstaller.cs

```diff
@@ -174,17 +174,17 @@ private void DeleteService()
             using (var serviceManagerHandle = new SafeServiceHandle(Interop.Advapi32.OpenSCManager(null, null, Interop.Advapi32.ServiceControllerOptions.SC_MANAGER_ALL)))
             {
                 if (serviceManagerHandle.IsInvalid)
-                    throw new Win32Exception("Could not open SCM");
+                    throw new Win32Exception($"Could not open SCM. {new Win32Exception().Message}");
```

Comment:
`Win32Exception` constructor that takes string reads the error code using `Marshal.GetLastPInvokeError()`. 

Formatting of the error message runs a lot of code that can modify `Marshal.GetLastPInvokeError()`. It means the outer Win32Exception can end up with a bogus error code.

This should manage the error codes explicitly.

Reply:
Yup. BTW should we have an API that gets the message for an error code in a cleaner way? Eg a static on Win32Exception.